pkg/querier/blocks_store_queryable.go

```diff
@@ -38,6 +38,13 @@ import (
 	"github.com/cortexproject/cortex/pkg/util/spanlogger"
 )
 
+const (
+	// The maximum number of times we retry fetching missing blocks from different
+	// store-gateways. If no more store-gateways are left (ie. due to lower replication
+	// factor) than we'll end the retries earlier.
+	maxFetchSeriesTries = 3
```

Comment:
I'd use word attempt here (and elsewhere in the code), but perhaps better to consult with native english speaker.